addons/account/models/account_bank_statement_line.py

```diff
@@ -227,13 +227,17 @@ def _compute_running_balance(self):
                 """,
                 [max_index, journal.id] + extra_params,
             )
+            pending_items = self
             for st_line_id, amount, is_anchor, balance_start, state in self._cr.fetchall():
                 if is_anchor:
                     current_running_balance = balance_start
                 if state == 'posted':
                     current_running_balance += amount
                 if record_by_id.get(st_line_id):
                     record_by_id[st_line_id].running_balance = current_running_balance
+                    pending_items -= record_by_id[st_line_id]
+            for item in pending_items:
+                item.running_balance = item.running_balance
```

Comment:
why this line?

Reply:
The `running_balance` field is compute not store, therefore, all records must set a value. 
Currently a select https://github.com/odoo/odoo/blob/16.0/addons/account/models/account_bank_statement_line.py#L190 is being run to set the values. If we delete a line in the bank payment form view we will see that no value is set: `odoo.exceptions.CacheMiss: 'account.bank.statement.line(<NewId origin=111>,).running_balance`